### ISSUE TITLE
Spectrogram: new plot type

### ISSUE THREAD

**Issue by swharden** [ScottPlot/ScottPlot#2319]
This issue tracks progress toward adding a tools (or at least documentation) making it easy to plot frequency data over time in the form of a spectrogram. I'll work on this some time after #2034 but I am creating the issue now so this goal is not forgotten.

Since I maintain the .NET Spectrogram package https://github.com/swharden/Spectrogram it seems important for me to improve the connection between these two projects

@yfzhn made good progress in this direction with #2259 but due to some complexities with that pull request I chose not to merge it at that time

**Comment by swharden** [comment]
I still like this idea, but it's low priority at the moment relative to the other work I need to complete.

There is a new FAQ page that describes how to create new plot types, and it may be useful for anyone who wants to create a spectrogram plot type in the future: https://scottplot.net/faq/custom-plot-type/
